docs/kernel-versions.md

```diff
@@ -0,0 +1,32 @@
+# BPF Features by Linux Kernel Version
+
+Major milestone releases: 4.1, 4.4.
+
+## 3.18
+
+- bpf syscall.
+
+## 3.19
+
+- socket support: bpf can attach to sockets.
+
+## 4.1
+
+- kprobe support: BPF programs can now instrument any kernel function via kernel dynamic tracing.
+
+## 4.3
+
+- debug string support: bpf_trace_printk() supports strings.
+
+## 4.4
+
+- bpf_perf_event_output: used by many tools that print per-event output. Eg, opensnoop.
+- unprivileged BPF: non-root usage.
```

Comment:
I'd add that it's for sockets only otherwise readers will get an impression that bpf+tracing can be unpriv as well.